src/Database/QueryCompiler.php

```diff
@@ -455,9 +466,28 @@ protected function _buildValuesPart(array $parts, Query $query, ValueBinder $bin
     protected function _buildUpdatePart(array $parts, Query $query, ValueBinder $binder): string
     {
         $table = $this->_stringifyExpressions($parts, $binder);
+        $hint = $this->_buildHintPart($query->clause('hint'), $query, $binder);
         $modifiers = $this->_buildModifierPart($query->clause('modifier'), $query, $binder);
 
-        return sprintf('UPDATE%s %s', $modifiers, implode(',', $table));
+        return sprintf('UPDATE%s%s %s', $hint, $modifiers, implode(',', $table));
+    }
+
+    /**
+     * Builds the optimizer hint comment part.
+     *
+     * @param list<string> $parts The optmizer hints
+     * @param \Cake\Database\Query $query The query that is being compiled
+     * @param \Cake\Database\ValueBinder $binder Value binder used to generate parameter placeholder
+     * @return string Optimizer hint comment
+     */
+    protected function _buildHintPart(array $parts, Query $query, ValueBinder $binder): string
+    {
+        $driver = $query->getConnection()->getDriver($query->getConnectionRole());
+        if ($parts === [] || !$driver->supports(DriverFeatureEnum::OPTIMIZER_HINT_COMMENT)) {
+            return '';
+        }
+
```

Comment:
```suggestion
        if ($parts === []) {
            return '';
        }
        $driver = $query->getConnection()->getDriver($query->getConnectionRole());
        if ($driver->supports(DriverFeatureEnum::OPTIMIZER_HINT_COMMENT)) {
            return '';
        }

```

Could we restructure this so that the empty case does less work to reduce the impact of the no-op path?

Reply:
Yes, I had planed a similar clean up.